bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -413,6 +413,14 @@ private List<String> botSpecificIntegrationBlockers(Map<Issue, Optional<IssueTra
     private void updateCheckBuilder(CheckBuilder checkBuilder, PullRequestCheckIssueVisitor visitor, List<String> additionalErrors) {
         if (visitor.isReadyForReview() && additionalErrors.isEmpty()) {
             checkBuilder.complete(true);
+            // It means whitespaceCheck failed as warning
+            if (!visitor.getAnnotations().isEmpty()) {
+                checkBuilder.title("Optional");
+                checkBuilder.summary("WhitespaceCheck failed as warning, it won't block the integration, but you'd better fix them.");
+                for (var annotation : visitor.getAnnotations()) {
+                    checkBuilder.annotation(annotation);
+                }
```

Comment:
I don't think we can assume that all `annotations` are whitespace warnings in this method. I don't think we need a summary text here, but it's hard to tell from just reading the code. The "Optional" title instead of "Required" and the list of annotations should be enough I think.

Reply:
I was trying to understand the role of the summary and it seems when there are errors, we are listing the 50 first errors in the summary. In GitHub, when creating a `check`, we are then adding the summary and each annotation as an `annotation`, so this makes sense. In GitLab, there is no concept of a `check` so we print in a comment, first the summary and then the annotations. This must result in duplicate listings for the 50 first whitespace errors in GitLab. That's what had me confused and why I thought the summary wasn't necessary.

So in this case, as you are pointing out, we need to put something there. How about:

```
"These warnings will not block integration"
```

Reply:
Sure, I will just use what you said, it's cleaner.